intern/frontend/src/index.jsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+import * as ReactDOM from 'react-dom'
+import { Cart } from './components/Cart/Cart'
+import { Shop } from './components/Shop/Shop'
+
+const App = () => {
+    const [products, setProducts] = React.useState([])
+    const [availableProducts, setAvailableProducts] = React.useState([])
+    const [inCartProducts, setInCartProducts] = React.useState([])
+
+    React.useEffect(() => {
+        fetch('http://localhost:8080/products')
+            .then((response) => response.json())
+            .then((products) => {
+                setProducts(products.products)
+            })
+    }, [])
+
+    React.useEffect(() => {
```

Comment:
maybe explain what is going on here too in a comment ?
explaining that whenever products state is set we want to set availableProducts and inCartProducts states and explain what these two state arrays represent